cytotable/warehouse/iceberg.py

```
@@ -528,21 +533,94 @@ def write_iceberg_warehouse(  # noqa: PLR0913
+            options are inconsistent (for example, ancillary image options
+            provided without ``image_dir``, ``image_dir``/``mask_dir``/
+            ``outline_dir`` referencing a missing directory, missing join SQL,
+            or ``page_keys`` lacking a non-empty ``'join'`` entry while image
+            export is requested).
+        ImportError:
+            Raised when the optional ``pyiceberg`` dependency is unavailable.
+        ValueError:
+            Raised when Iceberg export's join configuration is missing -- an
+            empty ``joins`` SQL string or a ``page_keys`` mapping without a
+            non-empty ``'join'`` entry.
+    """  # noqa: DOC503
```

Ruff (0.15.12)

[warning] Invalid rule code in `# noqa`: DOC503

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)